New test utility to mock HTTP requests/responses (#246)
EasyPost/easypost-php: README.md

- `USPS_CARRIER_ACCOUNT_ID` (eg: one-call buying a shipment for non-EasyPost employees)
- `PARTNER_USER_PROD_API_KEY` (eg: creating a referral user)
- `REFERRAL_USER_PROD_API_KEY` (eg: adding a credit card to a referral user)

#### Mocking

> Some of our unit tests require HTTP calls that cannot be easily tested with live/recorded calls (e.g. HTTP calls that

Reviewer: As mentioned before, please match the current styling of the document which is not to break up paragraphs on newlines if they are too long.

- `USPS_CARRIER_ACCOUNT_ID` (eg: one-call buying a shipment for non-EasyPost employees)
- `PARTNER_USER_PROD_API_KEY` (eg: creating a referral user)
- `REFERRAL_USER_PROD_API_KEY` (eg: adding a credit card to a referral user)

#### Mocking

Some of our unit tests require HTTP calls that cannot be easily tested with live/recorded calls (e.g. HTTP calls that
trigger payments or interact with external APIs).

We have implemented a custom, lightweight HTTP mocking functionality in this library that allows us to mock HTTP calls
and responses.

A mock client is the same as a normal client, with a set of mock request-response pairs stored as a property.

> At the time to make a real HTTP request, a mock client will instead check which mock request entry matches the queued

Reviewer: ```suggestion
At the time of making a real HTTP request, a mock client will instead check which mock request entry matches the queued
```

request (matching by HTTP method and a regex pattern for the URL), and will return the corresponding mock response (HTTP
status code and body).

**NOTE**: If a client is configured with a mocking utility, it will ONLY make mock requests. If it attempts to
make a request that does not match any of the configured mock requests, the request will fail and trigger an exception.

To use the mocking utility:

```php
use EasyPost\Test\mocking\MockingUtility;
use EasyPost\Test\mocking\MockRequest;
use EasyPost\Test\mocking\MockRequestMatchRule;

> use EasyPost\Test\mocking\MockRequestResponseInfo;

Reviewer: Must be `Mocking`
